Assets/Scripts/Enemies/Armadillo/Animation/AnimationState.cs

```diff
@@ -0,0 +1,47 @@
+using UnityEngine;
+
+namespace Enemies.Armadillo.Animation {
+    public class IdleAnimation : AnimationState {
+        public IdleAnimation(Animator animator) : base("ArmadilloIdle", animator) {
+        }
+    }
+
+    public class WalkAnimation : AnimationState {
+        public WalkAnimation(Animator animator) : base("ArmadilloWalk", animator) {
+        }
+    }
+
+    public class RushAnimation : AnimationState {
+        public RushAnimation(Animator animator) : base("ArmadilloRush", animator) {
+        }
+    }
+
+    public class StunAnimation : AnimationState {
+        public StunAnimation(Animator animator) : base("ArmadilloStun", animator) {
+        }
+    }
+
+    /** TODO: This is a common utility class and should be moved. */
+    public abstract class AnimationState : IState {
```

Comment:
This is a lot of boilerplate that's only necessary because `StateMachine` groups transitions by the `GetType()` of the states. If `StateMachine` could group states differently (maybe by hashcode or ID?) we wouldn't need these subclasses.